src/raftstore/store/peer_storage.rs

```diff
@@ -306,7 +306,11 @@ impl PeerStorage {
 
         let key = try!(SnapKey::from_snap(snap));
         let snap_file = try!(self.snap_mgr.rl().get_snap_file(&key, false));
-        defer!(self.snap_mgr.wl().deregister(&key, false));
+        self.snap_mgr.wl().register(key.clone(), SnapEntry::Applying);
```

Comment:
Then it will be always cloned inside `register`. It's better to let outside decide whether clone or move.